backend/server/routers/planner.py

```diff
@@ -289,19 +289,14 @@ def unschedule(data: CourseCode, token: str = DUMMY_TOKEN):
         token (str, optional): The user's authentication token. Defaults to DUMMY_TOKEN.
     """
     user = get_user(token)
-    planner = user['planner']
 
-    removed: Set[str] = set()
-    # Remove every instance of the course from each year
-    for year in planner['years']:
+    # Remove every instance of the course from each year and add to unplanned
+    for year in user['planner']['years']:
         for course_list in year.values():
```

Comment:
nitpick: probably just inline `year`; its not used again and is probably more readable than the nesting

```suggestion
for course_list in user['planner']['years'].values():
```